apps/api/src/app/events/usecases/map-trigger-recipients/map-trigger-recipients.use-case.ts

```diff
@@ -27,7 +27,7 @@ interface ILogTopicSubscribersPayload {
 }
 
 const isNotTopic = (recipient: TriggerRecipientSubscriber): recipient is TriggerRecipientSubscriber =>
-  typeof recipient === 'string' || recipient?.type !== TriggerRecipientsTypeEnum.TOPIC;
+  typeof recipient === 'string' || !recipient.hasOwnProperty('type');
```

Comment:
What's the reason behind changing this? We might not be using the type for a different situation of a Topic, but we wanted to leave that way open for future solutions. I can totally see us allowing to trigger an event to all the subscribers of a TenantId (becoming `TriggerRecipientsTypeEnum.TENANT`) as related to the work we are doing on that side.

Reply:
@p-fernandez Because I removed the unused ['record' fields](https://github.com/novuhq/novu/pull/3938/files#diff-9d8bab12d2fa35e0af04328affef14329e2db0a4715728bf2b955b25cac6ff89L17) from subscriber type (it was never used or saved anywhere), so in the context of this function - which subscriber is of type `TriggerRecipientSubscriber` - type script no longer accepts that there could be type property for subscriber. It wasn't exactly right even before, as it accepted it by 'mistake' 

We could refactor the function to be of different type/ different check when we need to and get to a specific case. Wdyt? makes sense?